addons/website_sale_delivery/views/website_sale_delivery_templates.xml

```diff
@@ -40,7 +40,7 @@
     </template>
 
     <template id="payment_delivery" name="Delivery Costs" inherit_id="website_sale.payment">
-        <xpath expr="//t[@t-if='website_sale_order.amount_total']" position="before">
+        <xpath expr="//t[@name='website_sale_non_free_cart']" position="before">
```

Comment:
You cannot xpath a node that you are defining at the same time in a different module in stable. If customers update the `website_sale_delivery` module but not `website_sale`, it will crash.

Reply:
Sorry, I thought the stable policy didn't apply in that case. But I get it.

Sorry for the dumb question but what is be the "good" R&D way to handle this? I'm changing the "`t-if`" condition, so this xpath will break. Should I add a "or" in the xpath to keep the previous one ? Or use a different xpath selector that works on both, targeting the parent or the first child or whatever?

Reply:
Honestly, the only good answer is "whatever works" :grimacing: 

It can get as ugly as you like, as long as it's cleaned in the forward-port to master.

Reply:
Great! I chose an ugly but simple

```
<!-- //t[@t-if='website_sale_order.amount_total'] should be removed in master -->
<xpath expr="//t[@name='website_sale_non_free_cart'] | //t[@t-if='website_sale_order.amount_total']" position="before">
```
and i will remove the `//t[@t-if='website_sale_order.amount_total']` part in master